WordPress/src/test/java/org/wordpress/android/ui/notifications/NotificationsListViewModelTest.kt

```diff
@@ -0,0 +1,206 @@
+        whenever(note.canModerate()).thenReturn(false)
+        whenever(readerPostTableWrapper.getBlogPost(siteId, postId, false)).thenReturn(readerPost)
+
+        // When
+        viewModel.openNote(noteId, action::openInTheReader, action::openDetailView)
+
+        // Then
+        verify(action, times(1)).openInTheReader(siteId, postId, commentId)
+        verify(action, times(0)).openDetailView()
+    }
+
+    @Test
+    fun `WHEN the note is a comment that cannot be moderated and the reader post is retrieved THEN open in reader`() {
+        // Given
+        val noteId = "1"
+        val siteId = 1L
+        val postId = 2L
+        val commentId = 3L
+        val note = mock<Note>()
+        whenever(notificationsUtilsWrapper.getNoteById(noteId)).thenReturn(note)
+        whenever(note.siteId).thenReturn(siteId.toInt())
+        whenever(note.postId).thenReturn(postId.toInt())
+        whenever(note.commentId).thenReturn(commentId)
+        whenever(note.isCommentType).thenReturn(true)
+        whenever(note.canModerate()).thenReturn(false)
+        whenever(readerPostTableWrapper.getBlogPost(siteId, postId, false)).thenReturn(null)
+        whenever(readerPostActionsWrapper.requestBlogPost(any(), any(), any())).then {
+            (it.arguments[REQUEST_BLOG_LISTENER_PARAM_POSITION] as ReaderActions.OnRequestListener<*>)
+                .onSuccess(null)
+        }
+
+        // When
+        viewModel.openNote(noteId, action::openInTheReader, action::openDetailView)
+
+        // Then
+        verify(action, times(1)).openInTheReader(siteId, postId, commentId)
+        verify(action, times(0)).openDetailView()
+    }
+
+    @Test
+    fun `WHEN the comment note cannot be moderated and the reader post retrieval fails THEN open detail view`() {
+        // Given
+        val noteId = "1"
+        val siteId = 1L
+        val postId = 2L
+        val note = mock<Note>()
+        whenever(notificationsUtilsWrapper.getNoteById(noteId)).thenReturn(note)
+        whenever(note.siteId).thenReturn(siteId.toInt())
+        whenever(note.postId).thenReturn(postId.toInt())
+        whenever(note.isCommentType).thenReturn(true)
+        whenever(note.canModerate()).thenReturn(false)
+        whenever(readerPostTableWrapper.getBlogPost(siteId, postId, false)).thenReturn(null)
+        whenever(readerPostActionsWrapper.requestBlogPost(any(), any(), any())).then {
+            (it.arguments[REQUEST_BLOG_LISTENER_PARAM_POSITION] as ReaderActions.OnRequestListener<*>)
+                .onFailure(500)
+        }
+
+        // When
+        viewModel.openNote(noteId, action::openInTheReader, action::openDetailView)
+
+        // Then
+        verify(action, times(0)).openInTheReader(any(), any(), any())
+        verify(action, times(1)).openDetailView()
+    }
+
+    private class ActionHandler {
+        fun openInTheReader(siteId: Long, postId: Long, commentId: Long) {
+            println("openInTheReader($siteId, $postId, $commentId)")
```

Comment:
The `println` is not really used. May be I could just have suppressed the compiler warning which are manifested as errors 😅